### PR TITLE
Chromium

### LINKED ISSUE #27: Package chromium
This is a tracking ticket for native chromium on musl.

Currently, flatpak is our best bet.

Once we have chromium, we can tackle ungoogled-chromium and electron.

### PR DESCRIPTION
**Pull request by git-bruh**
WIP

- [x] Remove dbus, atk
- [x] Figure out lib needed to not crash when trying to use gtk theme
- [X] Successful musl build
- [X] GCC build (no clang/llvm)
- [x] Ungoogle

`kiss a patch /usr/bin/patch` must be run before build, ungoogled scripts need more functionality from patch

closes #27

### CODE REVIEW

File: community/chromium/build [hunk 1]
```diff
@@ -0,0 +1,120 @@
+#!/bin/sh -e
+
+kiss l musl && {
```
Comment (jedahan) [inline community/chromium/build:3]: mind using full form `l` -> `list` ?

Comment (git-bruh) [reply community/chromium/build:3]: yeah sure, need to fix up the packaging part of the build aswell

File: community/chromium/build [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+vaapi=false
+kiss l libva && vaapi=true
+
+conf="
+    enable_nacl=false
+    enable_nacl_nonsfi=false
+    enable_js_type_check=false
+    is_clang=false
+    is_debug=false
+    clang_use_chrome_plugins=false
```
Comment (noocsharp) [inline community/chromium/build:69]: I think

Comment (noocsharp) [inline community/chromium/build:69]: I think this can be removed, it doesn't do anything when compiled with gcc.

**Review by noocsharp — CHANGES_REQUESTED** [review]

File: community/chromium/build (hunk 2 above)
Comment (noocsharp) [reply community/chromium/build:69]: `clang_use_chrome_plugins` that is

File: community/chromium/build [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+for patch in chromium-*.patch; do
+    patch -p1 < "$patch"
+done
+
+# Avoid falling back to preprocessor mode when sources contain time macros
+kiss list ccache && export CCACHE_SLOPPINESS=time_macros
+
+# Force script incompatible with Python 3 to use /usr/bin/python2
+sed -i '1s|python$|&2|' third_party/dom_distiller_js/protoc_plugins/*.py
+
```
Comment (noocsharp) [inline community/chromium/build:18]: Putting
```
mkdir -p third_party/node/linux/node-linux-x64/bin
ln -s /usr/bin/node third_party/node/linux/node-linux-x64/bin/node
```
up here will do the same thing as the `chromium-system-node.patch`. This is how Arch does it, and its a bit less fragile.

Comment (git-bruh) [reply community/chromium/build:18]: Yeah that's how i was doing it earlier, but thought that the patch made the build file a little less ugly /messy :p, doesn't do much at all, just makes the python script return `/usr/bin/node`

Comment (noocsharp) [reply community/chromium/build:18]: Fair enough. IMO putting it in the build script is cleaner, but its your package :)

File: community/chromium/depends [hunk 4]
```diff
@@ -0,0 +1,33 @@
+alsa-lib
```
Comment (noocsharp) [inline community/chromium/depends:1]: `libXtst` should be added as a dependency.

It is used in `third_party/webrtc/modules/desktop_capture/linux/shared_x_display.cc`

Comment (noocsharp) [inline community/chromium/depends:1]: `libXtst` should be added as a dependency

It is used in `third_party/webrtc/modules/desktop_capture/linux/shared_x_display.cc`

**Review by noocsharp — CHANGES_REQUESTED** [review]

**Comment by jedahan** [comment]
Maybe the build script can early exit with an error if the wrong patch is selected?

Or maybe we can switch patch (using kiss alternatives) on make, then switch back to whatever it was on exit?

Or maybe just set an alias during the build?

**Pull request merged**